plugins/field-grid-dropdown/src/index.js

```diff
@@ -0,0 +1,117 @@
+/**
+ * @license
+ * Copyright 2020 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @fileoverview Grid dropdown field.
+ * @author kozbial@google.com (Monica Kozbial)
+ */
+
+import Blockly from 'blockly/core';
+
+/**
+ * Grid dropdown field.
+ */
+export class FieldGridDropdown extends Blockly.FieldDropdown {
+  /**
+   * Class for an grid dropdown field.
+   * @param {(!Array.<!Array>|!Function)} menuGenerator A non-empty array of
+   *     options for a dropdown list, or a function which generates these
+   *     options.
+   * @param {Function=} opt_validator A function that is called to validate
+   *    changes to the field's value. Takes in a language-neutral dropdown
+   *    option & returns a validated language-neutral dropdown option, or null
+   *    to abort the change.
+   * @param {Object=} opt_config A map of options used to configure the field.
+   *    See the [field creation documentation]{@link https://developers.google.com/blockly/guides/create-custom-blocks/fields/built-in-fields/dropdown#creation}
+   *    for a list of properties this parameter supports.
+   * @extends {Blockly.Field}
+   * @constructor
+   * @throws {TypeError} If `menuGenerator` options are incorrectly structured.
+   */
+  constructor(menuGenerator, opt_validator, opt_config) {
+    super(menuGenerator, opt_validator, opt_config);
+
+    // TODO: Set columns from constructor and use in CSS.
```

Comment:
Before merging this PR, make an issue for the remaining TODOs (as checkboxes in the issue) so that it's easier to track them.

Reply:
I added checkbox list to #373 and added reference in TODOs.